### ISSUE TITLE
If object has too long string, Format-List throws error

### ISSUE THREAD

**Issue by naminodarie** [PowerShell/PowerShell#15911]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [x] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```powershell
[pscustomobject]@{"Text"=[string]::new("ア", 1000)}| Format-List
```

### Expected behavior

```console
Text : アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアアア
       アアアアアア
```
```


### Actual behavior

```console
out-lineoutput: startIndex cannot be larger than length of string. (Parameter 'startIndex')                                                 16:40:57
```

### Error details

```console
Type           : System.Management.Automation.CmdletInvocationException
ErrorRecord    : 
    Exception             : 
        Type       : System.ArgumentOutOfRangeException
        Message    : startIndex cannot be larger than length of string. (Parameter 'startIndex')
        ParamName  : startIndex
        TargetSite : 
            Name          : Substring
            DeclaringType : string
            MemberType    : Method
            Module        : System.Private.CoreLib.dll
        StackTrace : 
   at System.String.Substring(Int32 startIndex, Int32 length)
   at Microsoft.PowerShell.Commands.Internal.Format.StringManipulationHelper.GenerateLinesWithoutWordWrap(DisplayCells displayCells, String val, Int32 firstLineLen, Int32 followingLinesLen)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteSingleLineHelper(String prependString, String line, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteProperty(Int32 k, String propertyValue, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteProperties(String[] values, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ListOutputContext.ProcessPayload(FormatEntryData fed)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessPayload(FormatEntryData fed, OutputContext c)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessObject(PSObject so)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
        Source     : System.Private.CoreLib
        HResult    : -2146233086
    CategoryInfo          : NotSpecified: (:) [out-lineoutput], ArgumentOutOfRangeException
    FullyQualifiedErrorId : System.ArgumentOutOfRangeException,Microsoft.PowerShell.Commands.OutLineOutputCommand
    InvocationInfo        : 
        MyCommand       : out-lineoutput
        HistoryId       : 7
        CommandOrigin   : Internal
    ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
    PipelineIterationInfo : 



TargetSite     : 
    Name          : Invoke
    DeclaringType : System.Management.Automation.Runspaces.PipelineBase, System.Management.Automation, Version=7.1.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
    MemberType    : Method
    Module        : System.Management.Automation.dll
StackTrace     : 
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at System.Management.Automation.Runspaces.Pipeline.Invoke()
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options)
Message        : startIndex cannot be larger than length of string. (Parameter 'startIndex')
InnerException : 
    Type       : System.ArgumentOutOfRangeException
    Message    : startIndex cannot be larger than length of string. (Parameter 'startIndex')
    ParamName  : startIndex
    TargetSite : 
        Name          : Substring
        DeclaringType : string
        MemberType    : Method
        Module        : System.Private.CoreLib.dll
    StackTrace : 
   at System.String.Substring(Int32 startIndex, Int32 length)
   at Microsoft.PowerShell.Commands.Internal.Format.StringManipulationHelper.GenerateLinesWithoutWordWrap(DisplayCells displayCells, String val, Int32 firstLineLen, Int32 followingLinesLen)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteSingleLineHelper(String prependString, String line, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteProperty(Int32 k, String propertyValue, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.ListWriter.WriteProperties(String[] values, LineOutput lo)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ListOutputContext.ProcessPayload(FormatEntryData fed)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessPayload(FormatEntryData fed, OutputContext c)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessObject(PSObject so)
   at Microsoft.PowerShell.Commands.Internal.Format.OutCommandInner.ProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
    Source     : System.Private.CoreLib
    HResult    : -2146233086
Source         : System.Management.Automation
HResult        : -2146233087
```

though indent is broken, no error is thrown in PowerShell 5.1.

### Environment data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

$OutputEncoding

Preamble          :
BodyName          : utf-8
EncodingName      : Unicode (UTF-8)
HeaderName        : utf-8
WebName           : utf-8
WindowsCodePage   : 1200
IsBrowserDisplay  : True
IsBrowserSave     : True
IsMailNewsDisplay : True
IsMailNewsSave    : True
IsSingleByte      : False
EncoderFallback   : System.Text.EncoderReplacementFallback
DecoderFallback   : System.Text.DecoderReplacementFallback
IsReadOnly        : True
CodePage          : 65001
```

### Visuals

![image](https://user-images.githubusercontent.com/32071278/129158024-c9b73db0-db04-4a5d-888d-27eb889cd150.png)

**Comment by iSazonov** [comment]
What if you use an ASCII character in your test?

**Comment by naminodarie** [comment]
This error seems to occur only with CJK charactor. 

```powershell
PS Z:\> [pscustomobject]@{"Text" = ("ア" * 200) } | Format-List

out-lineoutput: startIndex cannot be larger than length of string. (Parameter 'startIndex')
PS Z:\> [pscustomobject]@{"Text" = ("한" * 200) } | Format-List

out-lineoutput: startIndex cannot be larger than length of string. (Parameter 'startIndex')
PS Z:\> [pscustomobject]@{"Text" = ("圀" * 200) } | Format-List

out-lineoutput: startIndex cannot be larger than length of string. (Parameter 'startIndex')
PS Z:\> [pscustomobject]@{"Text" = ("ｱ" * 200) } | Format-List

Text : ｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱ
       ｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱｱ


PS Z:\> [pscustomobject]@{"Text" = ("A" * 200) } | Format-List

Text : AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
       AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA


PS Z:\> [pscustomobject]@{"Text" = ("a" * 200) } | Format-List

Text : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
       aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa


PS Z:\> [pscustomobject]@{"Text" = ("α" * 200) } | Format-List

Text : αααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααα
       αααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααααα


PS Z:\> [pscustomobject]@{"Text" = ("Д" * 200) } | Format-List

Text : ДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДД
       ДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДДД


PS Z:\> [pscustomobject]@{"Text" = ("⡧" * 200) } | Format-List

Text : ⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧
       ⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧⡧


PS Z:\> [pscustomobject]@{"Text" = ("ⶁ" * 200) } | Format-List

Text : ⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁ
       ⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁⶁ


PS Z:\> [pscustomobject]@{"Text" = ("😀" * 200) } | Format-List

Text : 😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀�😀😀😀😀😀😀😀😀😀�
       😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀�😀😀😀😀😀😀😀😀😀�
       😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀�😀😀😀😀😀😀😀😀😀�
       😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀😀


```

**Comment by kumamiko** [comment]
No error in PowerShell-7.2.0-preview.8
It seems that this bug has been solved in `7.2.0-preview.5`.
[#13551](https://github.com/PowerShell/PowerShell/issues/13551)

**Comment by iSazonov** [comment]
@kumamiko Thanks for your investigations! I close this as duplication of #13551. We can reopen if needed.
